File: Footballapi/Footballapi/Models/Player.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Footballapi.Models
{
    public class Player
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int PlayerId { get; set; }
        public string PlayerName { get; set; } = string.Empty;
        public int PlayerAge { get; set; }
        public string PlayerNationality { get; set; } = string.Empty;
        public int PlayerShirtNumber { get; set; }
        public int PlayerValue { get; set; }
    }

    public class PlayerWithTeamId : Player
    {
        public int TeamId { get; set; }
    }

    public class PlayerWithTeam : Player
    { 
        public required Team Team {  get; set; }    
    }
}

File: FootballAngular/src/app/models/player.ts
import { Team } from './team';

export class Player {
  playerId: number;
  playerName: string;
  playerAge: number;
  playerNationality: string;
  playerShirtNumber: number;
  playerValue: number;

  constructor() {
    this.playerId = 0;
    this.playerName = '';
    this.playerAge = 0;
    this.playerNationality = '';
    this.playerShirtNumber = 0;
    this.playerValue = 0;
  }
}

export class PlayerWithTeamId extends Player {
  teamId: number;
  constructor() {
    super();
    this.teamId = 0;
  }
}

export interface PlayerWithTeam extends Player {
  team: Team;
}
